src/core/ledger/ledger.js

```diff
@@ -768,6 +774,79 @@ export class Ledger {
     account.payoutAddresses.delete(JSON.stringify(currencyResult.value));
   }
 
+  /**
+   * no-ops if integration tracking is already enabled
+   */
+  enableIntegrationTracking(): Ledger {
+    if (!this._trackIntegration) {
+      this._createAndProcessEvent({
+        type: "TOGGLE_INTEGRATION",
+      });
+    }
+    return this;
+  }
+
+  /**
+   * no-ops if integration tracking is already disabled
+   */
+  disableIntegrationTracking(): Ledger {
+    if (this._trackIntegration) {
+      this._createAndProcessEvent({
+        type: "TOGGLE_INTEGRATION",
+      });
+    }
+    return this;
+  }
+  _toggleIntegrationTracking(_: ToggleIntegration) {
+    this._trackIntegration = !this._trackIntegration;
+    if (!this._trackIntegration) this._integrationDistributions.clear();
+  }
+
+  /**
+   * Mark integrations as executed.
+   * Validates that:
+   * a) executions are tracked
+   * b) executions are not yet marked as executed
+   * c) distribution tracking is enabled
+   */
+  runIntegration(id: DistributionId): Ledger {
```

Comment:
`markDistributionExecuted` / `MARK_DISTRIBUTION_EXECUTED` - I don't imagine integrations should be called by the ledger in the future. Ideally there is an orchestrator at the API level running integrations and then writing them to ledger. So the execution will never be the ledger event; the marking of the execution is the ledger event. And the integration isn't being marked, then distribution is.